test/integration/local/batching/publish-batch.test.js

```diff
@@ -0,0 +1,39 @@
+const publishBatch = require('../../../../app/batching/publish-batch')
+const { BlobServiceClient } = require('@azure/storage-blob')
+const config = require('../../../../app/config').storageConfig
+const streamToBuffer = require('../../../stream-to-buffer')
+let blobServiceClient
+let outboundContainer
+
+describe('publish batch', () => {
+  beforeAll(async () => {
+    blobServiceClient = BlobServiceClient.fromConnectionString(config.connectionStr)
+    outboundContainer = blobServiceClient.getContainerClient(config.outboundContainer)
+    await outboundContainer.deleteIfExists()
+  })
+  test('should generate batch with correct filename', async () => {
+    const filename = 'PFELM0001_AP_20210827134400 (SITI).csv'
+    const content = [['Vendor'], ['Ledger']]
+    await publishBatch(filename, content)
+
+    const fileList = []
+    for await (const item of outboundContainer.listBlobsFlat()) {
+      fileList.push(item.name)
+    }
+    expect(fileList.filter(x => x === filename).length).toBe(1)
+  })
+
+  test('should generate batch with correct content', async () => {
+    const filename = 'PFELM0001_AP_20210827134400 (SITI).csv'
+    const content = [['Vendor'], ['Ledger']]
+    await publishBatch(filename, content)
+
+    const blobClient = outboundContainer.getBlobClient(filename)
+    const downloadResponse = await blobClient.download()
```

Comment:
You can use `blobClient.downloadToBuffer()` so you won't need to convert the stream to a buffer yourself

Reply:
That is good and it pleases me.  However, the reason I did it this way is because it matches the approach in the docs.

https://docs.microsoft.com/en-us/javascript/api/overview/azure/storage-blob-readme?view=azure-node-latest

I shall update.